include/circt/Dialect/ESI/cosim/svdpi.h

```diff
@@ -0,0 +1,380 @@
+ */
+
+#ifndef CIRCT_DIALECT_ESI_COSIM_SVDPI_H
+#define CIRCT_DIALECT_ESI_COSIM_SVDPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Define size-critical types on all OS platforms. */
+#if defined(_MSC_VER)
+typedef unsigned __int64 uint64_t;
+typedef unsigned __int32 uint32_t;
+typedef unsigned __int8 uint8_t;
+typedef signed __int64 int64_t;
+typedef signed __int32 int32_t;
+typedef signed __int8 int8_t;
+#elif defined(__MINGW32__)
+#include <stdint.h>
+#elif defined(__APPLE__)
+#include <stdint.h>
+#elif defined(__linux) || (defined(__APPLE__) && defined(__MACH__))
+#include <inttypes.h>
+#else
+#include <sys/types.h>
+#endif
+
+/* Use to import a symbol into dll */
+#ifndef DPI_DLLISPEC
+#if (defined(_MSC_VER) || defined(__MINGW32__) || defined(__CYGWIN__))
+#define DPI_DLLISPEC __declspec(dllimport)
+#else
+#define DPI_DLLISPEC
+#endif
+#endif
+
+/* Use to export a symbol from dll */
+#ifndef DPI_DLLESPEC
+#if (defined(_MSC_VER) || defined(__MINGW32__) || defined(__CYGWIN__))
+#define DPI_DLLESPEC __declspec(dllexport)
+#else
+#define DPI_DLLESPEC
+#endif
+#endif
+
+/* Use to mark a function as external */
+#ifndef DPI_EXTERN
+#define DPI_EXTERN
+#endif
+
+#ifndef DPI_PROTOTYPES
+#define DPI_PROTOTYPES
+/* object is defined imported by the application */
+#define XXTERN DPI_EXTERN DPI_DLLISPEC
+/* object is exported by the application */
+#define EETERN DPI_EXTERN DPI_DLLESPEC
+#endif
+
+/* canonical    representation */
+#define sv_0 0
+#define sv_1 1
+#define sv_z 2
+#define sv_x 3
+
+/* common type for 'bit' and 'logic' scalars. */
+typedef uint8_t svScalar;
+typedef svScalar svBit;   /* scalar */
+typedef svScalar svLogic; /* scalar */
+
+/*
+ * DPI representation of packed arrays.
+ * 2-state and 4-state vectors, exactly the same as PLI's avalue/bvalue.
+ */
+#ifndef VPI_VECVAL
+#define VPI_VECVAL
+typedef struct TVpiVecval {
+  uint32_t aval;
+  uint32_t bval;
+} s_vpi_vecval, *p_vpi_vecval;
+#endif
+
+/* (a chunk of) packed logic array */
+typedef s_vpi_vecval svLogicVecVal;
+
+/* (a chunk of) packed bit array */
+typedef uint32_t svBitVecVal;
+
+/* Number of chunks required to represent the given width packed array */
+#define SV_PACKED_DATA_NELEMS(WIDTH) (((WIDTH) + 31) >> 5)
+
+/*
+ * Because the contents of the unused bits is undetermined,
+ * the following macros can be handy.
+ */
+#define SV_MASK(N) (~(-1 << (N)))
+
+#define SV_GET_UNSIGNED_BITS(VALUE, N)                                         \
+  ((N) == 32 ? (VALUE) : ((VALUE)&SV_MASK(N)))
+
+#define SV_GET_SIGNED_BITS(VALUE, N)                                           \
+  ((N) == 32 ? (VALUE)                                                         \
+             : (((VALUE) & (1 << (N))) ? ((VALUE) | ~SV_MASK(N))               \
+                                       : ((VALUE)&SV_MASK(N))))
+
+/*
+ * Implementation-dependent representation.
+ */
+/*
+ * Return implementation version information string ("1800-2005" or "SV3.1a").
+ */
```

Comment:
This all seems like DPI generic interface logic, I think it should be in a top level lib/DPI library instead of being ESI specific.